frontend/src/common/navbar/index.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import AppBar from "@mui/material/AppBar";
+import Box from "@mui/material/Box";
+import CssBaseline from "@mui/material/CssBaseline";
+import Divider from "@mui/material/Divider";
+import Drawer from "@mui/material/Drawer";
+import IconButton from "@mui/material/IconButton";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemText from "@mui/material/ListItemText";
+import MenuIcon from "@mui/icons-material/Menu";
+import Toolbar from "@mui/material/Toolbar";
+import Typography from "@mui/material/Typography";
+import Button from "@mui/material/Button";
+
+interface Props {
+  navItems: string[];
+}
+
+const drawerWidth = 240;
+
+const NavBar = ({ navItems }: Props) => {
+  // const window = new Window();
+  const [mobileOpen, setMobileOpen] = React.useState(false);
+
+  const handleDrawerToggle = () => {
+    setMobileOpen((prevState) => !prevState);
+  };
+
+  const drawer = (
```

Comment:
It's better to define components outside of components (does that make sense? lol).
Can lead to weird bugs to rerenders.

Reply:
I agree, but just FYI this was code copied from MUI. I think this applies however this is not a component inside a component, but an Element inside a component (component = React FC & element = dom node if I'm correct). Either way I made the change, let me know if it's all good now.